gemma-core/src/main/java/ubic/gemma/model/expression/experiment/ExpressionExperimentValueObject.java

```diff
@@ -93,41 +90,34 @@ public ExpressionExperimentValueObject( ExpressionExperiment ee ) {
             this.taxonId = ee.getTaxon().getId();
         }
 
-        // AD
-        // FIXME: row[10] contains the taxon common name!
-        // Object technology = row[10];
-        // if ( technology != null ) {
-        //     this.technologyType = technology.toString();
-        // }
-
-        // 12-15 used in call to super
-
         // Counts
-        this.bioAssayCount = ee.getNumberOfSamples();
-        //  this.arrayDesignCount = ( ( Long ) row[17] ).intValue();
-        //  this.bioMaterialCount = ( ( Long ) row[19] ).intValue();
+        if ( ee.getBioAssays() != null && Hibernate.isInitialized( ee.getBioAssays() ) ) {
+            this.bioAssayCount = ee.getBioAssays().size();
+            // this is available too because AD are eagerly fetched
+            this.arrayDesignCount = ( int ) ee.getBioAssays().stream().map( BioAssay::getArrayDesignUsed ).count();
+        } else {
+            // this is a denormalization, so we merely use it as a fallback if bioAssays are not initialized
+            this.bioAssayCount = ee.getNumberOfSamples();
+            this.arrayDesignCount = null; // the number of AD is unknown, unfortunately
+        }
 
-        // Other
-        if ( ee.getExperimentalDesign() != null ) {
+        // ED
+        if ( ee.getExperimentalDesign() != null && Hibernate.isInitialized( ee.getExperimentalDesign() ) ) {
```

Comment:
ED are lazy-proxy by default, so we cannot assume that they are readily available.